### PR TITLE
Allow plugin to be inherited

### PR DESCRIPTION
**Pull request by markrekveld**
- [ ] All [tests](https://github.com/jfrog/artifactory-maven-plugin#testing-the-plugin) passed. If this feature is not already covered by the tests, I added new tests.

This change allows the plugin to be configured in a parent pom and be applied without the need to add the plugin to its build definition.
It also wraps any existing `ExecutionListener` in the `MavenExecutionRequest` so that also get executed. This is especially useful for the `org.apache.maven.cli.event.ExecutionEventLogger`.

### CODE REVIEW

**Comment by yahavi** [comment]
recheckcla

**Comment by markrekveld** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by markrekveld** [comment]
recheckcla

File: src/main/java/org/jfrog/buildinfo/deployment/BuildInfoRecorder.java [hunk 1]
```diff
@@ -48,12 +49,14 @@
...
     private final Log logger;
+    private final ExecutionListener wrappedListener;
 
-    public BuildInfoRecorder(MavenSession session, Log logger, ArtifactoryClientConfiguration conf) {
+    public BuildInfoRecorder(MavenSession session, Log logger, ArtifactoryClientConfiguration conf, ExecutionListener wrappedListener) {
         this.buildInfoBuilder = new BuildInfoModelPropertyResolver(logger, session, conf);
         this.buildDeployer = new BuildDeployer(logger);
         this.logger = logger;
         this.conf = conf;
+        this.wrappedListener = wrappedListener;
```
Comment (yahavi) [inline src/main/java/org/jfrog/buildinfo/deployment/BuildInfoRecorder.java:59]: I think using this logic will help us to remove 16 times of `if (wrappedListener != null) {`:
```suggestion
        this.wrappedListener = ObjectUtils.defaultIfNull(wrappedListener, new AbstractExecutionListener());
```

**Review by yahavi — CHANGES_REQUESTED** [review]
Thanks for this contribution, @markrekveld!
1. Please look at my comment in the constructor. With that, I think we can change all code that looks like this:
```java
if (wrappedListener != null) {
    wrappedListener.mojoSucceeded(event);
}
```
To This:
```java
wrappedListener.mojoSucceeded(event);
```

2. Since we now override all of the methods and we mustn't miss anything. Lets change the `BuildInfoRecorder` inheritance to implementation that obligating overriding all methods: 
```java
public class BuildInfoRecorder implements BuildInfoExtractor<ExecutionEvent>, ExecutionListener {
```

File: src/main/java/org/jfrog/buildinfo/deployment/BuildInfoRecorder.java (hunk 1 above)
Comment (markrekveld) [reply src/main/java/org/jfrog/buildinfo/deployment/BuildInfoRecorder.java]: I updated the code to use this `ObjectUtils.defaultIfNull` approach and made the changes to all the `wrappedListener` calls

**Review by yahavi — APPROVED** [review]
Thanks, @markrekveld

**Comment by markrekveld** [comment]
Your welcome @yahavi  and thanks for merging the PR!

**Pull request merged**